de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/ui/tree/ContactPopMenu.java

```diff
@@ -38,49 +42,108 @@
 
     private final ContactTreeRootNode.ContactInfo contactInfo;
 
-    public ContactPopMenu(ContactTreeRootNode.ContactInfo contactInfo) {
+    ContactPopMenu(ContactTreeRootNode.ContactInfo contactInfo) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=238&issues=AWYa7GRAvOWp7Yd62c1O&open=AWYa7GRAvOWp7Yd62c1O)

[ISSUE_KEY]: AWYa7GRAvOWp7Yd62c1O